### PR TITLE
✨ Emphasize version diff that has breaking changes

### LINKED ISSUE #85: Emphasize updates with breaking changes assuming their versions as SemVer
Like this?

||Package|Before|After|
|---|---|---|---|
|⬆️| **siketyan/a** | **v0.0.0** | **v0.0.1** |
|⬆️| **siketyan/b** | **v0.1.0** | **v0.2.0** |
|⬆️| siketyan/c | v0.1.0 | v0.1.1 |
|⬆️| siketyan/d | v1.0.0 | v1.0.1 |
|⬆️| siketyan/e | v1.0.0 | v1.1.0 |
|⬆️| **siketyan/f** | **v1.0.0** | **v2.0.0** |

### PR DESCRIPTION
**Pull request by siketyan**
Closes #85

### CODE REVIEW

File: src/Versioning/CompatibilityAwareInterface.php [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+     *
+     * In Semantic Versioning (SemVer), first number of their version
+     * must be incremented after having a B.C. breaking changes generally.
+     * Therefore, SemVer can return the first number as a compatibility number.
+     *
+     * @return int
+     */
+    #[Pure]
+    public function getCompatibilityNumber(): int;
+}
```
Comment (mpyw) [inline src/Versioning/CompatibilityAwareInterface.php:24]: Smart interface ✨

File: src/Versioning/HasSemVerLikeCompatibility.php [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    public function getCompatibilityNumber(): int
+    {
+        // For 0.Y.* versions, uses Y as a compatibility number instead of X in X.*.* versions.
+        if ($this->getX() === 0) {
+            // To avoid returning the same number from 0.Y.* and X.*.*,
+            // inverts the number to minus here assuming both X and Y are positive.
+            return -$this->getY();
+        }
+
+        return $this->getX() * 10;
```
Comment (mpyw) [inline src/Versioning/HasSemVerLikeCompatibility.php:27]: Would you explain more about these parts?

Comment (siketyan) [reply src/Versioning/HasSemVerLikeCompatibility.php]: Sorry, it was my mistake.
I meant unstable 0.* versions must not have the same "compatibility number" as other stable versions, using negative int.
Now is that clear? [37728e2](https://github.com/siketyan/loxcan/pull/86/commits/37728e273cc74581d4968c2efc3b3c91fb027e7b)

Comment (siketyan) [reply src/Versioning/HasSemVerLikeCompatibility.php]: @mpyw

Comment (siketyan) [reply src/Versioning/HasSemVerLikeCompatibility.php]: [e96c256](https://github.com/siketyan/loxcan/pull/86/commits/e96c256203634dc396f78c39af4d15f1708f9b10)

**Review by mpyw — APPROVED** [review]

**Pull request merged**